### PR TITLE
feat: Update AWS SDK v2 to v3 [WIP]

### PR DESCRIPTION
**Pull request by davidvpe**
Hi all, I started working on this a while ago, but since I had encountered one road block that I wasn't able to upgrade yet. The authentication part. I hope this community can give me a hand to complete this PR and we can finally upgrade to v3

### CODE REVIEW

File: src/util/aws-util.ts [hunk 1]
```diff
@@ -81,28 +82,25 @@ export class AwsUtil {
...
+
+        // if no credentials(provider) was passed in this method, we use the default
+        const provider = credentials ?? defaultProvider({
+            profile: this.isPartition ? this.partitionProfile : this.profile,
+        });
+
+        // save the credentialsProvider for any of the services we initialize
+        AwsUtil.credentialsProvider = provider;
+
+        // oc: not sure if this is needed, my guess is not
```
Comment (rene84) [inline src/util/aws-util.ts:95]: I'm not sure. Isn't it necessary to pick up sso credentials for example?

> To do this, specify a credential process in the shared AWS conﬁg file or the shared credentials file. If the AWS_SDK_LOAD_CONFIG environment variable is set to a truthy value, the SDK will prefer the process specified in the config file over the process specified in the credentials file (if any).

File: src/util/aws-util.ts [hunk 2]
```diff
@@ -81,28 +82,25 @@ export class AwsUtil {
...
-            if (this.partitionProfile) {
-                process.env.AWS_SDK_LOAD_CONFIG = '1';
-                process.env.AWS_PROFILE = this.partitionProfile;
-            }
-            stsClient = new STSClient({ region: this.partitionRegion });
-        }
+    public static async Initialize(credentials?: Provider<AwsCredentialIdentity>): Promise<void> {
+
+        // if no credentials(provider) was passed in this method, we use the default
+        const provider = credentials ?? defaultProvider({
```
Comment (rene84) [inline src/util/aws-util.ts:88]: is this custom credentials provider strictly here for testing purposes? I can't find anywhere in the code where it is invoked with a customer credentials provider

File: src/util/aws-util.ts (hunk 1 above)
Comment (rene84) [reply src/util/aws-util.ts:95]: hmm.. the default credentials provider seems to be capable of dealing with all of it. we could just try to not set it and see what happens

File: src/util/aws-util.ts (hunk 2 above)
Comment (OlafConijn) [reply src/util/aws-util.ts:88]: we used to do that for GovCloud (see [here](https://github.com/org-formation/org-formation-cli/blob/416dbaaf6f531a76278d8226850336a4398eec58/src/util/aws-util.ts#L96C12-L101C16))

**Pull request closed without merging**